kuma/wiki/views/document.py

```diff
@@ -865,106 +826,56 @@ def react_document(request, document_slug, document_locale):
-        elif original_doc.parent_id and original_doc.parent.locale == 'en-US':
-            en_slug = original_doc.parent.slug
-        else:
-            en_slug = ''
-
-        share_text = ugettext(
-            'I learned about %(title)s on MDN.') % {"title": doc.title}
-
-        contributors = doc.contributors
-        contributors_count = len(contributors)
-        has_contributors = contributors_count > 0
-        other_translations = original_doc.get_other_translations(
-            fields=['title', 'locale', 'slug', 'parent']
-        )
-        all_locales = (set([original_doc.locale]) |
-                       set(trans.locale for trans in other_translations))
-
-        # Bundle it all up and, finally, return.
-        context = {
-            'document': original_doc,
-            'document_html': doc_html,
-            'toc_html': toc_html,
-            'quick_links_html': quick_links_html,
-            'body_html': body_html,
-            'contributors': contributors,
-            'contributors_count': contributors_count,
-            'contributors_limit': 6,
-            'has_contributors': has_contributors,
-            'fallback_reason': fallback_reason,
-            'kumascript_errors': ks_errors,
-            'macro_sources': (
-                kumascript.macro_sources(force_lowercase_keys=True)
-                if ks_errors else
-                None
-            ),
-            'render_raw_fallback': rendering_params['render_raw_fallback'],
-            'seo_summary': seo_summary,
-            'seo_parent_title': seo_parent_title,
-            'share_text': share_text,
-            'search_url': get_search_url_from_referer(request) or '',
-            'analytics_page_revision': doc.current_revision_id,
-            'analytics_en_slug': en_slug,
-            'content_experiment': rendering_params['experiment'],
-            'other_translations': other_translations,
-            'all_locales': all_locales,
-        }
-        response = render(request, 'wiki/react_document.html', context)
-
-    if ks_errors or request.user.is_authenticated:
-        add_never_cache_headers(response)
-
-    # We're doing this to prevent any unknown intermediate public HTTP caches
-    # from erroneously caching without considering cookies, since cookies do
-    # affect the content of the response. The primary CDN is configured to
-    # cache based on a whitelist of cookies.
-    patch_vary_headers(response, ('Cookie',))
+    contributors = doc.contributors
+    contributors_count = len(contributors)
+    has_contributors = contributors_count > 0
+    other_translations = doc.get_other_translations(
+        fields=['title', 'locale', 'slug', 'parent']
+    )
+    all_locales = (set([doc.locale]) |
+                   set(trans.locale for trans in other_translations))
+
+    # Bundle it all up and, finally, return.
+    context = {
+        'document_api_data': base64.b64encode(
+            json.dumps(document_api_data(doc))),
```

Comment:
This is how I pass the JSON blob to the frontend when the document is first loaded, so that the frontend doesn't have to make a fetch() call to get it. The base64 encoding is because the data is going to go in an HTML <script> tag, and the data could include the string "</script>" which would screw things up.